circuit_passes/src/passes/loop_unroll/extracted_location_updater.rs

```diff
@@ -44,80 +47,210 @@ impl ExtractedFunctionLocationUpdater {
+                        id: new_id(),
+                        source_file_id: bucket_meta.get_source_file_id().clone(),
+                        line: bucket_meta.get_line(),
+                        message_id: bucket_meta.get_message_id(),
+                        context: InstrContext { size: 1 },
+                        dest_is_output: false,
                         dest_address_type: AddressType::SubcmpSignal {
-                            cmp_address: build_u32_value(bucket, arena),
+                            cmp_address: build_u32_value(bucket_meta, *arena),
                             uniform_parallel_value: None,
                             counter_override: false,
                             is_output: false,
-                            //TODO: Not sure what to put here. If I put Unknown (assuming the later pass
-                            //  would correct) it crashes somewhere. What I really need is Last in the
-                            //  proper place to make it generate the *_run function at the right time
-                            //  but NoLast in locations prior to that (I think). Why isn't Unknown handled
-                            //  by the later pass deterministic subcomp pass or something? Always using
-                            //  Last here could result in the run function being called too soon.
-                            //SEE: circom/tests/subcmps/subcmps0C.circom
                             input_information: InputInformation::Input {
-                                status: StatusInput::Last,
+                                status: StatusInput::Last, // This is the key to generating call to "run" function
                             },
                         },
                         dest: LocationRule::Indexed {
-                            location: build_u32_value(bucket, 0), //the value here is ignored by the 'bounded_fn' below
-                            template_header: match &bucket.dest {
+                            location: build_u32_value(bucket_meta, 0), //the value here is ignored by the 'bounded_fn' below
+                            template_header: match dest {
                                 LocationRule::Indexed { template_header, .. } => {
                                     template_header.clone()
                                 }
                                 LocationRule::Mapped { .. } => todo!(),
                             },
                         },
-                        src: build_u32_value(bucket, 0), //the value here is ignored at runtime
-                        bounded_fn: Some(String::from(FR_IDENTITY_ARR_PTR)), //NOTE: doesn't have enough arguments but it works out
+                        src: build_u32_value(bucket_meta, 0), //the value here is ignored at runtime
+                        bounded_fn: Some(String::from(LLVM_DONOTHING_FN_NAME)), // actual result ignored, only need effect of 'StatusInput::Last'
```

Comment:
This `StoreBucket` exists only to force the "run" function to be generated at the end of `StoreBucket::produce_llvm_ir`. In that same function, if `bounded_fn` were `None`, it would end up generating an actual store instruction or a call to `fr_copy_n` but both of those would be undesirable because we only need to generate the call to the "run" function. This quirk exists because of the weird nature of the Circom IR, specifically how the decision to generate a call to the "run" function is implicitly encoded within the `StoreBucket` rather than having an explicit bucket to represent that construct.

Reply:
Gotcha. Maybe we can add a TODO/make a ticket to refactor this a little at some point to avoid this quirkiness.

Reply:
That would cut deep into original circom code so I think Kostas would want to avoid making those changes.